### PR TITLE
regions plugin: add maxRegions option to limit max nr of created regions

### PR DESCRIPTION
**Pull request by osheroff**
I've been heavily using the regions plugin in a project i'm playing with.  One feature I definitely needed (for my UI's sanity) was to limit the number of regions a user could create (to 1, in my case).

### CODE REVIEW

File: src/plugin/regions.js [hunk 1]
```diff
@@ -643,6 +643,7 @@ export default class RegionsPlugin {
         this.params = params;
         this.wavesurfer = ws;
         this.util = ws.util;
+        this.maxRegions = params.maxRegions;
```
Comment (thijstriemstra) [inline src/plugin/regions.js]: can you also document this at the top of this class, where the other params are documented?

File: src/plugin/regions.js [hunk 2]
```diff
@@ -836,7 +837,11 @@ export default class RegionsPlugin {
                 return;
             }
 
+
             if (!region) {
+                if (this.maxRegions && Object.keys(this.list).length >= this.maxRegions)
```
Comment (thijstriemstra) [inline src/plugin/regions.js]: add a comment describing what this if statement does.

File: src/plugin/regions.js (hunk 1 above)
Comment (osheroff) [reply src/plugin/regions.js]: as I'm documenting this I realize the implementation I wrote only honors maxRegions when a user tries to create a region.  Wondering if it should be honored when a region is programmatically created....

**Comment by osheroff** [comment]
documented

File: src/plugin/regions.js (hunk 1 above)
Comment (thijstriemstra) [reply src/plugin/regions.js]: >  Wondering if it should be honored when a region is programmatically created....

I would assume it does..

Comment (osheroff) [reply src/plugin/regions.js]: ok, maxRegions is now honored from the api too..

**Pull request merged**